workshops/NeuralNetFromScratch/README.md

```diff
@@ -0,0 +1,237 @@
+```
+ 
+And we will always need the derivative of our activation function:
+```python
+def sigmoidDerivative(x):
+    return x * (1 - x)
+```
+The derivative will be later multiplied by the error when calculating the adjustment values.
+
+For those of you who are not fond of calculus, all the "derivative" does essentially is find the slope of any function.
+ 
+Now let's create a neuron in the function which we'll call `base`
+```python
+def base(inputs):
+    inputs = inputs.astype(float) 
+    return sigmoid(np.dot(inputs, synapticWeights))
+```
+ 
+# Train
+This is the most important part in building a neural network, here, we will create the `train` function.
+ 
+Within the `train` function, we will input a few arguments. These arguments include: training inputs, training outputs, and training iterations.
+ 
+You might be asking, "Why do we include the outputs? Isn't that what the NN has to figure out itself?" 
+ 
+The reason why we include the outputs is because we need the correct outputs in training so we can calculate the error.
+ 
+ 
+```python
+def train(trainingInputs, trainingOutputs, trainingIterations):
+    global synapticWeights
+    for iteration in range(trainingIterations):
+        
+        output = base(trainingInputs)
+        
+        error = trainingOutputs - output
+        
+        adjustments = np.dot(trainingInputs.T, error * sigmoidDerivative(output))
+ 
+        synapticWeights += adjustments
+```
+ 
+As you can see in the function above, `error = training_outputs - output ` calculates the error.
+ 
+This line here, `adjustments = np.dot(training_inputs.T, error * sigmoid_derivative(output))` calculates the adjustments to the weights (The backpropagation I was talking about)
+ 
+Which you can see change here: `synaptic_weights += adjustments`
+
+# What is BackPropagation?
+Backpropagation is the process of tweaking the weights, remember how the weights were random in the beginning? Well, now we are artificially changing the weight values.
+
+<img src="https://hackclub.slack.com/files/U010VMPCEBF/F01DLN09RFS/backprop.png?origin_team=T0266FRGM&origin_channel=D010V8S08G4" alt="Back Propagation"/>
+ 
+ 
+# Training it
+Now let's actually train our model!
+ 
+In the following lines of code, I will create training inputs and transpose them.
+ 
+```python
+trainingInputs = np.array([[0,0,1],
+                            [1,1,1],
+                            [1,0,1],
+                            [0,1,1]])
+ 
+trainingOutputs = np.array([[0,1,1,0]]).T
+```
+ 
+Then I will pass this through the train function:
+```python
+train(trainingInputs, trainingOutputs, 10000)
+```
+ 
+# Get The Results!
+ 
+We will prompt the user for inputs:
+```python
+ 
+input1 = str(input("Input 1: "))
+input2 = str(input("Input 2: "))
+input3 = str(input("Input 3: "))
+```
+ 
+And finally we will pass it through our network!!!
+```python
+base(np.array([input1, input2, input3]))
+```
+ 
+```
+input 1: 1
+input 2: 0
+input 3: 0
+```
+Output:
```

Comment:
You should include a step explaining how to run the program and what they should expect